app/src/main/java/org/oppia/app/player/exploration/ExplorationFragmentPresenter.kt

```diff
@@ -33,7 +50,24 @@ class ExplorationFragmentPresenter @Inject constructor(
     }
     return binding
   }
+  private fun getProfileData(): LiveData<StoryTextSize> {
+    return Transformations.map(
+      profileManagementController.getProfile(profileId),
+      ::processGetProfileResult
+    )
+  }
 
+  private fun subscribeToAudioLanguageLiveData() {
+    getProfileData().observe(activity, Observer<StoryTextSize> { result ->
+      FontScaleConfigurationUtil.adjustFontSize(activity,result)
```

Comment:
Please add tests for this integration pathway.

Reply:
Will be adding test after confirming the best approach as mentioned in explanations.